airflow/ui/src/pages/DagsList/Dag/Tasks/TaskRecentRuns.tsx

```diff
@@ -0,0 +1,97 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Box, Text } from "@chakra-ui/react";
+import { Flex } from "@chakra-ui/react";
+import dayjs from "dayjs";
+import duration from "dayjs/plugin/duration";
+
+import type { TaskInstanceResponse } from "openapi/requests/types.gen";
+import Time from "src/components/Time";
+import { Tooltip } from "src/components/ui";
+import { stateColor } from "src/utils/stateColor";
+
+dayjs.extend(duration);
+
+const BAR_HEIGHT = 60;
+
+export const TaskRecentRuns = ({
+  taskInstances,
+}: {
+  readonly taskInstances: Array<TaskInstanceResponse>;
+}) => {
+  if (!taskInstances.length) {
+    return undefined;
+  }
+
+  const taskInstancesWithDuration = taskInstances.map((taskInstance) => ({
+    ...taskInstance,
+    duration:
+      dayjs
+        .duration(dayjs(taskInstance.end_date).diff(taskInstance.start_date))
+        .asSeconds() || 0,
+  }));
+
+  const max = Math.max.apply(
+    undefined,
+    taskInstancesWithDuration.map((taskInstance) => taskInstance.duration),
+  );
+
+  return (
+    <Flex alignItems="flex-end" flexDirection="row-reverse">
+      {taskInstancesWithDuration.map((taskInstance) =>
+        taskInstance.state === null ? undefined : (
+          <Tooltip
+            content={
+              <Box>
+                <Text>State: {taskInstance.state}</Text>
+                <Text>
+                  Start Date: <Time datetime={taskInstance.start_date} />
+                </Text>
+                <Text>
+                  End Date: <Time datetime={taskInstance.end_date} />
+                </Text>
+                <Text>Try Number: {taskInstance.try_number}</Text>
```

Comment:
Let's only show try_number if it's greater than 1

Reply:
Thanks, done.